### PR TITLE
Add option to utilize html-minifier in rendering process

### PR DESCRIPTION
**Pull request by ericrglass**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -170,7 +180,11 @@ function fastifyView (fastify, opts, next) {
       if (!this.res.getHeader('content-type')) {
         this.header('Content-Type', 'text/html; charset=' + charset)
       }
-      this.send(toHtml(data))
+      if (options.useHtmlMinifier && (typeof options.useHtmlMinifier.minify === 'function')) {
+        this.send(options.useHtmlMinifier.minify(toHtml(data), options.htmlMinifierOptions || {}))
```
Comment (mcollina) [inline index.js]: I think this should not be needed here, as the minifier is already applied in `readCallback` before putting it in the cache. Can you confirm/check?

File: index.js [hunk 2]
```diff
@@ -249,7 +272,11 @@ function fastifyView (fastify, opts, next) {
...
```
Comment (mcollina) [inline index.js]: same here

File: README.md [hunk 3]
```diff
@@ -114,6 +114,27 @@ To use partials in mustache you will need to pass the names and paths in the opt
   }
 ```
 
+To utilize [`html-minifier`](https://www.npmjs.com/package/html-minifier) in the rendering process, except for the 'pug' engine and streams, you can add the option useHtmlMinifier with a reference to html-minifier,
```
[the remark this answers is not in the document]
Comment (ericrglass) [reply README.md]: I really do not know the pug template engine, and had troubles getting the unit test to work. Will try again. I was not sure how to implement for template engines that support streams. Will look into that further too.

File: index.js (hunk 1 above)
Comment (ericrglass) [reply index.js]: Will remove and try the unit test again to verify that it is not necessary.

File: index.js (hunk 2 above)
Comment (ericrglass) [reply index.js]: Same here too about removing and retesting.

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (ericrglass) [reply README.md]: Changes have been made to support pug and streams, and unit test and docs have been updated.

File: index.js (hunk 1 above)
Comment (ericrglass) [reply index.js]: Completed

File: index.js (hunk 2 above)
Comment (ericrglass) [reply index.js]: Completed

File: README.md [hunk 4]
```diff
@@ -114,6 +114,47 @@ To use partials in mustache you will need to pass the names and paths in the opt
   }
 ```
 
+To utilize [`html-minifier`](https://www.npmjs.com/package/html-minifier) in the rendering process, you can add the option useHtmlMinifier with a reference to html-minifier,
```
Comment (mcollina) [inline README.md]:
```suggestion
To utilize [`html-minifier`](https://www.npmjs.com/package/html-minifier) in the rendering process, you can add the option `useHtmlMinifier` with a reference to `html-minifier`,
```

File: README.md [hunk 5]
```diff
@@ -114,6 +114,47 @@ To use partials in mustache you will need to pass the names and paths in the opt
...
+ and the optional htmlMinifierOptions option is used to specify the html-minifier options:
```
Comment (mcollina) [inline README.md]:
```suggestion
 and the optional `htmlMinifierOptions` option is used to specify the `html-minifier` options:
```

File: README.md [hunk 6]
```diff
@@ -114,6 +114,47 @@ To use partials in mustache you will need to pass the names and paths in the opt
...
+  removeEmptyAttributes: true
+}
+// in template engine options configure the use of html-minifier
+  options: {
+    useHtmlMinifier: minifier,
+    htmlMinifierOptions: minifierOpts
+  }
+```
+To utilize [`html-minify-stream`](https://www.npmjs.com/package/html-minify-stream) in the rendering process with template engines that support streams,
+ you can add the option useHtmlMinifyStream with a reference to html-minify-stream, and the optional htmlMinifierOptions option is used to specify the options just like html-minifier:
```
Comment (mcollina) [inline README.md]:
```suggestion
 you can add the option `useHtmlMinifyStream` with a reference to `html-minify-stream`, and the optional `htmlMinifierOptions` option is used to specify the options just like `html-minifier`:
```

**Comment by mcollina** [comment]
Do you think this should be backported to the 2.x line (for Fastify v1)? master is for the upcoming fastify v2 release.

File: README.md (hunk 5 above)
Comment (ericrglass) [reply README.md]: Thanks for the suggestions

**Comment by ericrglass** [comment]
I have made changes to package.json and index.js; and re-tested, so that this feature can now be used with the current fastify version.

**Comment by mcollina** [comment]
Unfortunately this was not exactly what I asked. The code was good as it was before, I asked if it should be backported. In that case, another PR against https://github.com/fastify/point-of-view/tree/2.x might be filled.

**Comment by ericrglass** [comment]
Sorry for the misunderstanding, I have reverted my fork of the master branch back to be compatible with master branch 3.x. I will make a backport for the 2.x branch now.

**Pull request merged**